Everything's looking good, except on TPU backends some of the `testKde` tests fail with e.g.:
```
AssertionError: 
Not equal to tolerance rtol=0.001, atol=0.001

Mismatched elements: 7 / 50 (14%)
Max absolute difference: 0.07452869
Max relative difference: 0.00547456
 x: array([ -6.917634,  -8.133598,  -7.125779,  -7.454879,  -7.296328,
        -7.365776,  -6.817926,  -9.488862,  -7.272069,  -8.460855,
       -13.688169,  -6.750305,  -6.494191,  -7.032907,  -8.428143,...
 y: array([ -6.916273,  -8.131803,  -7.12294 ,  -7.452182,  -7.291719,
        -7.362315,  -6.817333,  -9.481265,  -7.267247,  -8.457088,
       -13.613641,  -6.748472,  -6.494314,  -7.032794,  -8.427602,...
```
Can you set the rtol to 0.01 if `device_under_test` is `"tpu"`?

--- a/tests/scipy_stats_test.py
+++ b/tests/scipy_stats_test.py
@@ -34,12 +34,12 @@ one_and_two_dim_shapes = [(4,), (3, 4), (3, 1), (1, 4)]
 
 
 def genNamedParametersNArgs(n):
-    return parameterized.named_parameters(
-        jtu.cases_from_list(
-          {"testcase_name": jtu.format_test_name_suffix("", shapes, dtypes),
-            "shapes": shapes, "dtypes": dtypes}
-          for shapes in itertools.combinations_with_replacement(all_shapes, n)
-          for dtypes in itertools.combinations_with_replacement(jtu.dtypes.floating, n)))
+  return parameterized.named_parameters(
+      jtu.cases_from_list(
+        {"testcase_name": jtu.format_test_name_suffix("", shapes, dtypes),
+          "shapes": shapes, "dtypes": dtypes}
+        for shapes in itertools.combinations_with_replacement(all_shapes, n)
+        for dtypes in itertools.combinations_with_replacement(jtu.dtypes.floating, n)))
 
 
 # Allow implicit rank promotion in these tests, as virtually every test exercises it.
@@ -704,7 +704,7 @@ class LaxBackedScipyStatsTests(jtu.JaxTestCase):
       for func in [None, "evaluate", "logpdf", "pdf"]))
   def testKde(self, inshape, dtype, outsize, weights, method, func):
     if method == "callable":
-        method = lambda kde: jax.numpy.power(kde.neff, -1./(kde.d+4))
+      method = lambda kde: jax.numpy.power(kde.neff, -1./(kde.d+4))
 
     def scipy_fun(dataset, points, w):
       w = np.abs(w) if weights else None
@@ -732,8 +732,11 @@ class LaxBackedScipyStatsTests(jtu.JaxTestCase):
     rng = jtu.rand_default(self.rng())
     args_maker = lambda: [
       rng(inshape, dtype), rng(outshape, dtype), rng(inshape[-1:], dtype)]
-    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker,
-                            tol={np.float32: 1e-3, np.float64: 1e-14})
+    self._CheckAgainstNumpy(
+        scipy_fun, lax_fun, args_maker, tol={
+            np.float32: 1e-2 if jtu.device_under_test() == "tpu" else 1e-3,
+            np.float64: 1e-14
+        })
     self._CompileAndCheck(
         lax_fun, args_maker, rtol={np.float32: 3e-07, np.float64: 4e-15})
 
@@ -887,4 +890,4 @@ class LaxBackedScipyStatsTests(jtu.JaxTestCase):
     self.assertAllClose(evaluate_kde(kde, x), kde.evaluate(x))
 
 if __name__ == "__main__":
-    absltest.main(testLoader=jtu.JaxTestLoader())
+  absltest.main(testLoader=jtu.JaxTestLoader())

